src/video/uikit/SDL_uikitmetalangle.h

```diff
@@ -0,0 +1,44 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2021 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#ifndef SDL_uikitmetalangle_
+#define SDL_uikitmetalangle_
+
+#if SDL_VIDEO_METALANGLE
+
+#include "../SDL_sysvideo.h"
+
+extern int UIKit_GL_MakeCurrent(_THIS, SDL_Window * window,
```

Comment:
These files have a lot of duplicate code. I wonder if it might be cleaner to merge it all into `SDL_uikitopengles` etc. with a couple #ifdefs

Reply:
Uhm, there's a lot of duplicated code (especially for the header files), but, instead `src/video/uikit/SDL_uikitmetalangleview.m` is going to be definitely cleaner into a separate file.

I will try to merge the two implementations when changing `SDL_VIDEO_METALANGLE` to `SDL_VIDEO_OPENGL_METALANGLE` unless it makes sense to have both on the same file.